script.js

```diff
@@ -1,4 +1,608 @@
+    playerTurn
+  )}`;
+  if (autoGenerate == 0) {
+    message += `<br><br>Choose the order of the dice (1 or 2).`;
+  } else if (autoGenerate == 1) {
+    message += `<br><b>Auto-Generate Mode</b>: Press '<b>Submit</b>' to continue...`;
+  }
+  gamePhase = 2;
+  return message;
+};
+
+// Function: Game Phase 1 (Roll - Variable Number of Dice)
+var phase1VariableDice = function () {
+  var counter = 0;
+  var message = `<b>Player ${playerTurn}</b> has rolled... `;
+  playerRolls.push(String(playerTurn));
+  while (counter < numOfDice) {
+    var roll = rollDice(6);
+    playerRolls.push(roll);
+    message += `<br>Dice ${counter + 1}: ${roll}`;
+    counter += 1;
+  }
+  message += `<br><br>Press '<b>Submit</b>' to see optimal combined number.`;
+  gamePhase = 2;
+  return message;
+};
+
+// Function: Game Phase 1 (Roll - Knockout)
+var phase1Knockout = function () {
+  knockoutGone.push(playerTurn);
+  var message = `<b>Player ${playerTurn}</b>, you're up!<br><br>`;
+  playerRolls.push(String(playerTurn));
+  playerRolls.push(rollDice(6));
+  playerRolls.push(rollDice(6));
+  var roll1 = playerRolls[(knockoutRound - 1) * 3 + 1];
+  var roll2 = playerRolls[(knockoutRound - 1) * 3 + 2];
+  message += `You rolled <b>${roll1}</b> for Dice 1 and <b>${roll2}</b> for Dice 2.<br><br>Choose the order of the dice (1 or 2).`;
+  gamePhase = 2;
+  return message;
+};
+
+// Function: Game Phase 2 (Choice - Standard)
+var phase2 = function (choice) {
+  finalNumbers.push(chosenOrderNum(choice, playerRolls));
+  var message = `<b>Player ${playerTurn}</b>, you chose Dice ${choice} first.<br><br>Your number is <b>${
+    finalNumbers[playerTurn - 1]
+  }</b>`;
+  return message;
+};
+
+// Function: Game Phase 2 (Choice - Variable Number of Dice)
+var phase2VariableDice = function () {
+  var index = 0;
+  if (gameMode == "Highest Mode") {
+    var number = 6;
+    var highestNum = "";
+    while (number > 0) {
+      index = (playerTurn - 1) * (numOfDice + 1) + 1;
+      while (
+        index > (playerTurn - 1) * (numOfDice + 1) &&
+        index < playerTurn * (numOfDice + 1)
+      ) {
+        if (playerRolls[index] == number) {
+          highestNum += playerRolls[index];
+        }
+        index += 1;
+      }
+      number -= 1;
+    }
+    finalNumbers.push(Number(highestNum));
+    return `Your highest number is <b>${highestNum}</b>`;
+  } else if (gameMode == "Lowest Mode") {
+    var number = 0;
+    var lowestNum = "";
+    while (number <= 6) {
+      index = (playerTurn - 1) * (numOfDice + 1) + 1;
+      while (
+        index > (playerTurn - 1) * (numOfDice + 1) &&
+        index < playerTurn * (numOfDice + 1)
+      ) {
+        if (playerRolls[index] == number) {
+          lowestNum += playerRolls[index];
+        }
+        index += 1;
+      }
+      number += 1;
+    }
+    finalNumbers.push(Number(lowestNum));
+    return `Your lowest number is <b>${lowestNum}</b>`;
+  }
+};
```

Comment:
quite complicated, my brain doesn't want to process it but good effort on your part